### PR TITLE
Validate submitted MBIDs

### PR DESCRIPTION
**Pull request by clinty**
Check to see if release_mbid, recording_mbid, or artist_mbids contain
any invalid UUIDs.

This should fix LB-74.

### CODE REVIEW

File: webserver/views/api.py [hunk 1]
```diff
@@ -253,7 +253,17 @@ def _validate_listen(listen):
                 if len(tag) > MAX_TAG_SIZE:
                     _log_raise_400("JSON document may not contain track_metadata.additional_info.tags "
                                    "longer than %d characters." % MAX_TAG_SIZE, listen)
-
+        # MBIDs
+        lmbid = listen['track_metadata']['additional_info'].get('release_mbid')
+        if not is_valid_uuid(lmbid):
+                _log_raise_400("Release MBID format invalid.")
```
Comment (alastair) [inline webserver/views/api.py]: Watch indent here

File: webserver/views/api.py [hunk 2]
```diff
@@ -253,7 +253,17 @@ def _validate_listen(listen):
...
+        cmbid = listen['track_metadata']['additional_info'].get('recording_mbid')
+        if not is_valid_uuid(cmbid):
+                _log_raise_400("Recording MBID format invalid.")
+        ambids = listen['track_metadata']['additional_info'].get('artist_mbids')
+        for ambid in ambids:
```
Comment (alastair) [inline webserver/views/api.py]: If `artist_mbids` isn't in additional_info then .get will return None, use `.get('artist_mbids', [])` here

**Comment by clinty** [comment]
better?

**Comment by mineo** [comment]
This would fix [LB-74](http://tickets.musicbrainz.org/browse/LB-74).

AFAICT this would make the recording and release mbids mandatory, but not the artist ones.

File: webserver/views/api.py [hunk 3]
```diff
@@ -253,7 +254,17 @@ def _validate_listen(listen):
                 if len(tag) > MAX_TAG_SIZE:
                     _log_raise_400("JSON document may not contain track_metadata.additional_info.tags "
                                    "longer than %d characters." % MAX_TAG_SIZE, listen)
-
+        # MBIDs
+        lmbid = listen['track_metadata']['additional_info'].get('release_mbid')
+        if not is_valid_uuid(lmbid):
```
Comment (gentlecat) [inline webserver/views/api.py]: It's probably better to check if this value is actually specified first (same with other MBID checks).

**Comment by clinty** [comment]
like that?

**Comment by gentlecat** [comment]
Looks good.

**Comment by alastair** [comment]
Thanks. I made a small change to the dict getting (so that we only have the full key once), but otherwise it's fine.

**Comment by 19379** [comment]
this does not work as expected. when supplying an invalid MBID, it throws a 500 internal server error...

`_log_raise_400() takes exactly 2 arguments (1 given)`

**Comment by alastair** [comment]
Yes, we noticed this (in our error logs). It will be fixed in the next
release

**Pull request merged**